src/UI/Component/Input/Field/Factory.php

```diff
@@ -205,4 +205,30 @@ public function dependantGroup(array $inputs);
 	 * @return    \ILIAS\UI\Component\Input\Field\Checkbox
 	 */
 	public function checkbox($label, $byline = null);
-}
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A select is used to allow users to pick among a number of options
+	 *   composition: >
+	 *     Select field will render a select-tag with a number of options.
+	 *   effect: >
+	 *     Only one option is selectable.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: Select Input MAY be used for choosing from predetermined options.
+	 *
+	 *   interaction:
+	 *     1: Only one option is selectable.
+	 *
+	 * ---
+	 * @param $options array with the select options as key-value pairs.
+	 * @param $label string defines the label.
+	 * @param $byline string
+	 * @return \ILIAS\UI\Component\Input\Field\Select
+	 */
+	public function select(array $options, $label, $byline = null);
```

Comment:
I would prefer `$label` as the first argument for consistencies sake. I would also like to briefly discuss the position of `$byline` in the argument list in the JF to get a consistent approach there. Also see [Tag Input](https://github.com/ILIAS-eLearning/ILIAS/pull/802/files#diff-2d8571e3a451f51b51f9868081f6f7e8R269).

Reply:
My first thought was also $label as first element, but when I saw the method "section" I decided to keep the same order to be consistent. 
`public function section(array $inputs, $label, $byline = null);`

Should I change the select method to this?
`public function select(array $options, $label, $byline = null);`

Reply:
Ah, k, we have another precedence there that I didn't notice. Maybe we should change `section` to `section($label, array $inputs, $byline = null)` or even `section($label, $byline, array $inputs)`? I would prefer to find common parameters ($label, $byline) in common positions for all inputs. I even would prefer to always have `$byline` as second parameter and remove the default to `null`. Thoughts?

Reply:
Ok, I will change the label and options order but, I don't have an strong opinion about where to place the $byline. I think could be better to keep it in the last position and avoid to send empty strings if the byline is not required. What do you think?

Reply:
JF: we prefer the following order: "section($label, array $inputs, $byline = null)".